### PR TITLE
Convert var.c to var.d

### PR DESCRIPTION
**Pull request by wilzbach**
As this is my first file in the D conversion, be a bit careful.

### CODE REVIEW

File: src/dmd/backend/var.d [hunk 1]
```diff
@@ -147,68 +166,68 @@ int level = 0;                  /* declaration level                    */
...
 {
-    #define X(a,b)      b,
-        ENUMSCMAC
-    #undef X
-};
 
-#endif /* !SPP */
-volatile int controlc_saw;              /* a control C was seen         */
-symtab_t globsym;               /* global symbol table                  */
+char[SCMAX] sytab;
```
Comment (wilzbach) [inline src/dmd/backend/var.d]: This is part of https://github.com/dlang/dmd/pull/8421 too, hence it's probably a good idea to merge https://github.com/dlang/dmd/pull/8421 first.

File: src/dmd/backend/var.d [hunk 2]
```diff
@@ -47,25 +51,29 @@ int TYdelegate = TYllong;
...
+        {
+            TYnfunc,
+            TYnpfunc,
+            TYnpfunc,
+            TYnfunc,
+        };
+    }
+else
+{
+tym_t[LINK_MAXDIM][MEMMODELS] functypetab =
```
Comment (jacob-carlborg) [inline src/dmd/backend/var.d]: Shouldn't `LINK_MAXDIM` and `MEMMODELS` be flipped?

**Comment by WalterBright** [comment]
Never mind, I'm working on it.

**Comment by WalterBright** [comment]
Rebooted as #8735

**Pull request closed without merging**